### PR TITLE
Add TemporalPauseInfo search attribute

### PR DESCRIPTION
**Pull request by ychebotarev**
## What changed?
Add PausedInfo search attribute

## Why?
To allow users find workflows with paused activities

## How did you test it?
make unit-test.

### CODE REVIEW

**Comment by bergundy** [comment]
We need to be very mindful of which search attributes we add to the system. I think we need to figure out how to fit all types of pauses into this search attribute. We want to support pausing other primitives and via different policies.
Removing a policy wouldn’t necessarily mean that an activity of a specific type should be unpaused.

Another minor note is that all system search attributes should now be prefixed with `system.` and searchable without the prefix.

Let’s discuss offline.

File: common/searchattribute/defs.go
Comment (rodrigozhou) [inline common/searchattribute/defs.go:86]:
```suggestion
	// PausedActivityTypes stores the list of paused activity types in the workflow.
	// Note: there can be multiple activities with the same type but different activity IDs. In this case there will be only one entry in the list.
```

Suppose there are 2 (or more) activities paused, all of the same type. If the user only resume one of the activities, how are you keeping track if there are still activities paused from the same type?

File: schema/elasticsearch/visibility/versioned/v8/index_template_v7.json
Comment (rodrigozhou) [inline schema/elasticsearch/visibility/versioned/v8/index_template_v7.json:1]: Do not create the `v8` folder. This version corresponds to the Elasticsearch version, not the schema version.

File: schema/mysql/v8/visibility/schema.sql
Comment (rodrigozhou) [inline schema/mysql/v8/visibility/schema.sql:47]: `PauseActivityTypes` -> `PausedActivityTypes`

I see this typo in several places, please fix them.

**Review by rodrigozhou — CHANGES_REQUESTED** [review]

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go:86]: Every time user pause/unpause activity, I will "rebuild" PausedActivityTypes from pending activities. 
In this scenario, if user has 2 paused activities of the same type, and one of them was unpaused:
I will go over pending activities, found those that are paused (there still be one), take its type and then build new PausedActivityTypes.
Since it is the same type, search attribute will not change and we will not update it.

File: schema/elasticsearch/visibility/versioned/v8/index_template_v7.json
Comment (ychebotarev) [reply schema/elasticsearch/visibility/versioned/v8/index_template_v7.json]: removed v8 folder

File: schema/mysql/v8/visibility/schema.sql
Comment (ychebotarev) [reply schema/mysql/v8/visibility/schema.sql:47]: Renamed to PausedEntities everywhere

File: common/searchattribute/defs.go
Comment (bergundy) [inline common/searchattribute/defs.go:88]: Maybe call this `primitive_type` and `entity_type` because seems like we'll be using activity type and workflow type in this field?

File: common/searchattribute/defs.go
Comment (bergundy) [inline common/searchattribute/defs.go:91]: Need to figure how to express the new convention of `system.` search attributes.

File: common/searchattribute/defs.go
Comment (rodrigozhou) [reply common/searchattribute/defs.go]: Wouldn't it potentially be slow? Imagine a scenario with several workflows with several pending activities.

File: common/searchattribute/defs.go
Comment (rodrigozhou) [reply common/searchattribute/defs.go:91]: The `system.` prefix is gonna be a separate work, and it should work seamlessly with the existing search attributes as well. I think it will be part of the query converter when parsing the query. I don't think we need to worry about it in this PR.

File: common/searchattribute/defs.go
Comment (rodrigozhou) [inline common/searchattribute/defs.go:87]: Maybe create an enum type, and reference it here.

File: schema/mysql/v8/visibility/schema.sql
Comment (rodrigozhou) [inline schema/mysql/v8/visibility/schema.sql:47]: Rename: `s/PauseActivityTypes/PausedEntities`
You missed the renaming in a few places, please fix them.

File: schema/elasticsearch/visibility/versioned/v7/index_template_v7.json
Comment (rodrigozhou) [inline schema/elasticsearch/visibility/versioned/v7/index_template_v7.json:97]: Ugh. I'm sorry, you did the right thing before by creating the `v8` folder. I totally got confused.

**Review by rodrigozhou — CHANGES_REQUESTED** [review]

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go]: It will not be slow. 
To unpause I need access to mutable state either way. If I have access - it becomes a matter of going through the list of pending activities. Alternative is to parse existing attribute, then remove paused one, then build it back. Alternative is slower.

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go:88]: You mean primitive_type  instead of entity_id ? 
`primitive` and `entity` are kind of very close in semantic. And `primitive` usually imply something basic, building block of something. I think the result will be less clear. 
Also for workflow it will be workflow_id or run_id. I don't think we should 'collapse' them (and we also can't).

File: schema/elasticsearch/visibility/versioned/v7/index_template_v7.json
Comment (ychebotarev) [reply schema/elasticsearch/visibility/versioned/v7/index_template_v7.json:97]: done

File: schema/mysql/v8/visibility/schema.sql
Comment (ychebotarev) [reply schema/mysql/v8/visibility/schema.sql:47]: done

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go:87]: done

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go]: irrelevant.

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go:88]: yes, we can't. there is not way to "unpause this specific activity if it was paused manually" query.

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go]: updated

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go:88]: (that said - it will be resolved in batch API)

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go]: we agree on adding "Temporal" prefix to the search attribute.

File: common/searchattribute/defs.go
Comment (bergundy) [inline common/searchattribute/defs.go:101]: nit: Should this be `PauseInfo` and not `PausedInfo`? Seems like a better way of phrasing it.

File: schema/postgresql/v12/visibility/versioned/v1.7/add_paused_info_search_attribute.sql
Comment (bergundy) [inline schema/postgresql/v12/visibility/versioned/v1.7/add_paused_info_search_attribute.sql:3]: nit: inconsistent with the newlines at the end of files.

**Review by bergundy — APPROVED** [review]
Approved with comments. The SQL schema LGTM but wondering if you can confirm that upgrade works as expected before merging.

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go:100]: it may be a policy, it may be manual. Ok, I will remove it.

File: common/searchattribute/defs.go
Comment (ychebotarev) [reply common/searchattribute/defs.go:85]: done

File: common/searchattribute/defs.go
Comment (rodrigozhou) [reply common/searchattribute/defs.go:88]: Can you explain how is it resolved in batch API? Also, I'm not sure if this format is good to represent the information you want.

File: schema/elasticsearch/visibility/versioned/v8/index_template_v7.json
Comment (rodrigozhou) [inline schema/elasticsearch/visibility/versioned/v8/index_template_v7.json:96]: Please fix typo here and other places.
```suggestion
      "TemporalPauseInfo": {
```

File: schema/postgresql/v12/visibility/versioned/v1.7/manifest.json
Comment (rodrigozhou) [inline schema/postgresql/v12/visibility/versioned/v1.7/manifest.json:4]: Update description to match others.

**Review by rodrigozhou — CHANGES_REQUESTED** [review]

File: schema/postgresql/v12/visibility/versioned/v1.7/manifest.json
Comment (ychebotarev) [reply schema/postgresql/v12/visibility/versioned/v1.7/manifest.json:4]: updated

File: proto/internal/temporal/server/api/enums/v1/workflow.proto
Comment (ychebotarev) [reply proto/internal/temporal/server/api/enums/v1/workflow.proto:50]: workflowEntity - entity related to workflow. I think it is fine to be specific.

File: schema/elasticsearch/visibility/versioned/v8/upgrade.sh
Comment (rodrigozhou) [inline schema/elasticsearch/visibility/versioned/v8/upgrade.sh:34]: Typo

**Review by rodrigozhou — CHANGES_REQUESTED** [review]

File: schema/elasticsearch/visibility/versioned/v8/upgrade.sh
Comment (ychebotarev) [reply schema/elasticsearch/visibility/versioned/v8/upgrade.sh]: fixed.

**Review by rodrigozhou — APPROVED** [review]

**Pull request merged**
